ibridges/irodsconnector/sync.py

```diff
@@ -0,0 +1,376 @@
+        return "/".join(path.split(os.sep))
+
+    objects=[]
+    collections=[]
+
+    for root, dirs, files in os.walk(path):
+        for file in files:
+            full_path=Path(root) / file
+            rel_path=str(full_path)[len(str(path)):].lstrip(os.sep)
+            if max_level is None or rel_path.count(os.sep)<max_level:
+                objects.append(FileObject(
+                    name=file,
+                    path=fix_local_path(rel_path),
+                    size=full_path.stat().st_size,
+                    checksum=None if ignore_checksum else _calc_checksum(full_path)))
+
+        collections.extend([FolderObject(
+                fix_local_path(str(Path(root) / dir)[len(str(path)):].lstrip(os.sep)),
+                len([x for x in Path(f"{root}{os.sep}{dir}").iterdir() if x.is_file()]),
+                len([x for x in Path(f"{root}{os.sep}{dir}").iterdir() if x.is_dir()])
+            )
+            for dir in dirs if max_level is None or dir.count(os.sep)<max_level-1])
+
+    return objects, collections
+
+def _get_irods_tree(coll: iRODSCollection,
+                    root: str = '',
+                    level: int = 0,
+                    max_level: Union[int, None] = None,
+                    ignore_checksum: bool = False):
+
+    root=coll.path if len(root)==0 else root
+
+    # chksum() (re)calculates checksum, call only when checksum is empty
+    objects=[FileObject(
+        x.name,
+        x.path[len(root):].lstrip('/'),
+        x.size,
+        None if ignore_checksum else (x.checksum if len(x.checksum)>0 else x.chksum()))
+        for x in coll.data_objects]
+
+    if max_level is None or level<max_level-1:
+        collections=[FolderObject(
+            x.path[len(root):].lstrip('/'),
+            len(x.data_objects),
+            len(x.subcollections)) for x in coll.subcollections]
+
+        for subcoll in coll.subcollections:
+            subobjects, subcollections=_get_irods_tree(
+                coll=subcoll,
+                root=root,
+                level=level+1,
+                max_level=max_level,
+                ignore_checksum=ignore_checksum
+                )
+            objects.extend(subobjects)
+            collections.extend(subcollections)
+    else:
+        collections=[]
+
+    return objects, collections
+
+def _create_irods_collections(session: Session,
+                              target: IrodsPath,
+                              collections: list[FolderObject],
+                              dry_run: bool,
+                              copy_empty_folders: bool):
+    if dry_run:
+        print("Will create collection(s):")
+
+    for collection in collections:
+        if collection.is_empty() and not copy_empty_folders:
+            continue
+
+        full_path=target / collection.path
+
+        if dry_run:
+            print(f"  {full_path} {collection.n_files}/{collection.n_folders}")
+        else:
+            _=create_collection(session, str(full_path))
```

Comment:
```suggestion
            create_collection(session, str(full_path))
```
We don't need the underscore, right?